File: src/main/java/com/skysoft/mixin/ItemEntityRenderStateMixin.java
package com.skysoft.mixin;

import com.skysoft.features.misc.DroppedItemScaleRenderState;
import net.minecraft.client.renderer.entity.state.ItemEntityRenderState;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.Unique;

@Mixin(ItemEntityRenderState.class)
public abstract class ItemEntityRenderStateMixin implements DroppedItemScaleRenderState {
    @Unique private float skysoftDroppedItemScale = 1.0F;
    @Unique @Override public float skysoftGetDroppedItemScale() { return skysoftDroppedItemScale; }
    @Unique @Override public void skysoftSetDroppedItemScale(float scale) { skysoftDroppedItemScale = scale; }
}

File: src/main/kotlin/com/skysoft/mixin/ItemEntityRenderStateMixin.kt
package com.skysoft.mixin

import com.skysoft.features.misc.DroppedItemScaleRenderState
import net.minecraft.client.renderer.entity.state.ItemEntityRenderState
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.Unique

@Mixin(ItemEntityRenderState::class)
abstract class ItemEntityRenderStateMixin : DroppedItemScaleRenderState {
    @field:Unique
    private var skysoftDroppedItemScale = 1f

    @Unique
    override fun skysoftGetDroppedItemScale(): Float = skysoftDroppedItemScale

    @Unique
    override fun skysoftSetDroppedItemScale(scale: Float) {
        skysoftDroppedItemScale = scale
    }
}
